### PR TITLE
LUA record documentation (version number, shared model)

### PR DESCRIPTION
**Pull request by Habbie**
### Short description
I enthousiastically merged #7869 without docs. This PR rectifies that situation.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [ ] compiled this code
- [ ] tested this code
- [x] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

File: docs/settings.rst [hunk 1]
```diff
@@ -596,7 +596,11 @@ Enables EDNS subnet processing, for backends that support it.
 -  Boolean
```
Comment (pieterlexis) [inline docs/settings.rst]: not a boolean :)

I'd say 'string', 'ne of: no, yes, shared'

Comment (Habbie) [reply docs/settings.rst]: amended

File: docs/lua-records/index.rst [hunk 2]
```diff
@@ -197,6 +197,20 @@ Note that to protect operators, support for LUA records must be enabled
 explicitly, either globally (``enable-lua-records``) or per zone
 (``ENABLE-LUA-RECORDS`` = 1).
 
+.. _lua-records-shared-state:
+
+Shared Lua state model
+----------------------
+
+The default mode of operation for LUA records is to create a fresh Lua state for every query that hits a LUA records.
```
Comment (franklouwers) [inline docs/lua-records/index.rst]: a LUA record~~s~~

File: docs/lua-records/index.rst [hunk 3]
```diff
@@ -197,6 +197,20 @@ Note that to protect operators, support for LUA records must be enabled
...
+This way, different LUA records cannot accidentally interfere with eachother, by leaving around global objects, or perhaps even deleting relevant functions.
```
Comment (franklouwers) [inline docs/lua-records/index.rst]: ~~eachother~~ each other

**Comment by Habbie** [comment]
suggestions from @franklouwers commited and squashed

**Comment by mnordhoff** [comment]
It creates (up to) one state per thread and you still wind up with a handful of states, right? It might be worth mentioning that explicitly, so someone doesn't think there's precisely one state **total** and try to make a global counter or something.

**Comment by Habbie** [comment]
> It creates (up to) one state per thread and you still wind up with a handful of states, right? It might be worth mentioning that explicitly, so someone doesn't think there's precisely one state **total** and try to make a global counter or something.

pushed

**Pull request merged**